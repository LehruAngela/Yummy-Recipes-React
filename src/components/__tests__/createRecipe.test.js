import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import CreateRecipe from '../createRecipe';

describe('CreateRecipe', () => {
    const params = {
        match: {
          params: {
            id: 1,
          },
        },
      };
    const wrapper = shallow(<CreateRecipe match={{ params }} />);
    const preventDefault = jest.fn();

    it('renders properly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        }); 
    
    it('handles CreateRecipe', () => {
        expect(wrapper.instance().handleCreateRecipe({ preventDefault }));
        });


})