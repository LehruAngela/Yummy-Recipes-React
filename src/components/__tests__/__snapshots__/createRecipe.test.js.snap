// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateRecipe renders properly 1`] = `
<div>
  <button
    className="btn my-2 my-sm-0 addbutton link"
    data-target="#exampleModalCenter"
    data-toggle="modal"
    type="button"
  >
    <i
      class="fas fa-plus-circle"
    />
     Add Recipe
  </button>
  <div
    aria-hidden="true"
    aria-labelledby="exampleModalCenterTitle"
    class="modal fade"
    id="exampleModalCenter"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="modal-dialog modal-dialog-centered"
      role="document"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <h4
            class="modal-title"
            id="exampleModalLongTitle"
          >
            Add Recipe
          </h4>
          <button
            aria-label="Close"
            class="close"
            data-dismiss="modal"
            id="closeRecipeModal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              Ã—
            </span>
          </button>
        </div>
        <div
          class="modal-body"
        >
          <form
            onSubmit={[Function]}
          >
            <div>
              <label>
                Recipe Name
              </label>
              <br />
              <input
                name="recipe_name"
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
            <div>
              <label
                for="ingredients"
              >
                Ingredients
              </label>
              <br />
              <textarea
                cols="50"
                name="ingredients"
                onChange={[Function]}
                rows="3"
                type="text"
                value=""
              />
            </div>
            <div>
              <label
                for="directions"
              >
                Directions
              </label>
              <br />
              <textarea
                cols="50"
                name="directions"
                onChange={[Function]}
                rows="3"
                type="text"
                value=""
              />
            </div>
            <div
              class="modal-footer"
            >
              <button
                class="btn btn-primary"
                type="submit"
              >
                Create Recipe
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
